### ISSUE TITLE
Weighted Cross Entropy not working on high imbalance data

### ISSUE THREAD

**Issue by brcsomnath** [jakeret/tf_unet#31]
I am trying to implement a binary segmentation with data of high imbalance. I am getting a true positive as zero and false negative is always 100%. I have modified the error_rate() function to print the same as follows.

    def error_rate(predictions, labels):
    
        y_ = labels[...,1]
        tp = 0
        fn = 0

        Np = 0
        Nn = 0

        for i in range(y_.shape[0]):
            for j in range(y_.shape[1]):
                for k in range(y_.shape[2]):
                    if y_[i][j][k]:
                        Np += 1
                        if predictions[i][j][k][1] > predictions[i][j][k][0]:
                            tp += 1
                    else:
                        Nn += 1
                        if predictions[i][j][k][1] < predictions[i][j][k][0]:
                            fn += 1

        if Np == 0:
            tp = -1
        else:
            tp = tp/Np

        return tp*100, (fn/Nn*100)

I am using cross-entropy loss and I have tried with various class weights [0.01, 0.99], [1, 100]. Kindly provide some insight into what might be going wrong. The dataset I am using has sliced CBCT images and every image has labels of both classes present.

**Comment by jakeret** [comment]
A problem with such a high class imbalance is very hard to tackle. Actually, I'm not sure if this can be addressed with a weighted loss function. 
Have you tried to use the `dice_coefficient`? Most likely you have to resample the data or create synthetic data sets.
